src/shogun/structure/GEMPLP.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2014 Jiaolong Xu
+ * Copyright (C) 2014 Jiaolong Xu
+ */
+
+#ifndef __GEMPLP_H__
+#define __GEMPLP_H__
+
+#include <shogun/lib/config.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/structure/FactorGraph.h>
+#include <shogun/structure/Factor.h>
+#include <shogun/structure/MAPInference.h>
+#include <shogun/lib/SGNDArray.h>
+
+#include <vector>
+#include <set>
+
+using namespace std;
+
+namespace shogun
+{
+#define IGNORE_IN_CLASSLIST
+/** GEMPLP (Generalized Max-product LP Relaxation) inference for fatcor graph
+ *
+ * Please refer to following paper for more details:
+ *
+ * [1] Fixing max-product: Convergent message passing algorithms for MAP LP-relaxations
+ * Amir Globerson, Tommi Jaakkola
+ * Advances in Neural Information Processing Systems (NIPS). Vancouver, Canada. 2007.
+ *
+ * [2] Approximate Inference in Graphical Models using LP Relaxations.
+ * David Sontag
+ * Ph.D. thesis, Massachusetts Institute of Technology, 2010.
+ *
+ * The original implementation of GEMPLP can be found:
+ * http://cs.nyu.edu/~dsontag/code/mplp_ver2.tgz
+ * http://cs.nyu.edu/~dsontag/code/mplp_ver1.tgz
```

Comment:
What does it mean to follow at least 80%? Is it that 80% of the lines are copy & paste but not 100% of them? :-)